src/components/fields/multiline-text-field/multiline-text-field.spec.js

```diff
@@ -1,189 +1,181 @@
-      expect(textInput).toHaveProp('name', props.name);
-      expect(textInput).toHaveProp('value', props.value);
-      expect(textInput).toHaveProp('onChange', props.onChange);
-      expect(textInput).toHaveProp('onBlur', props.onBlur);
-      expect(textInput).toHaveProp('onFocus', props.onFocus);
-      expect(textInput).toHaveProp('isAutofocussed', props.isAutofocussed);
-      expect(textInput).toHaveProp('isDefaultClosed', props.isDefaultClosed);
-      expect(textInput).toHaveProp('isDisabled', props.isDisabled);
-      expect(textInput).toHaveProp('isReadOnly', props.isReadOnly);
-      expect(textInput).toHaveProp('placeholder', props.placeholder);
-      expect(textInput).toHaveProp('hasError', true);
-
-      expect(wrapper).toRender(FieldErrors);
-      expect(wrapper.find(FieldErrors)).toHaveProp('errors', props.errors);
-    });
+describe('when `badge` is passed', () => {
+  it('should render a badge', () => {
+    const { getByText } = renderMultilineTextField({ badge: 'foo badge' });
+    expect(getByText('foo badge')).toBeInTheDocument();
   });
+});
 
-  describe('when disabled', () => {
-    let props;
-    let wrapper;
-    beforeEach(() => {
-      props = createTestProps({ isDisabled: true });
-      wrapper = shallow(<MultilineTextField {...props} />);
-    });
-    it('should disable the MultilineTextInput', () => {
-      expect(wrapper.find(MultilineTextInput)).toHaveProp('isDisabled', true);
-    });
+describe('when disabled', () => {
+  it('should disable the input', () => {
+    const { getByLabelText } = renderMultilineTextField({ isDisabled: true });
+    expect(getByLabelText('MultilineTextField')).toHaveAttribute('disabled');
   });
+});
 
-  describe('when read-only', () => {
-    let props;
-    let wrapper;
-    beforeEach(() => {
-      props = createTestProps({ isReadOnly: true });
-      wrapper = shallow(<MultilineTextField {...props} />);
-    });
-    it('should mark the MultilineTextInput as read-only', () => {
-      expect(wrapper.find(MultilineTextInput)).toHaveProp('isReadOnly', true);
-    });
+describe('when readOnly', () => {
+  it('should disable the input', () => {
+    const { getByLabelText } = renderMultilineTextField({ isReadOnly: true });
+    expect(getByLabelText('MultilineTextField')).toHaveAttribute('readonly');
   });
+});
 
-  describe('when input should be closed by default', () => {
-    let props;
-    let wrapper;
-    beforeEach(() => {
-      props = createTestProps({ isDefaultClosed: true });
-      wrapper = shallow(<MultilineTextField {...props} />);
-    });
-    it('should mark the MultilineTextInput as default closed', () => {
-      expect(wrapper.find(MultilineTextInput)).toHaveProp(
-        'isDefaultClosed',
-        true
-      );
-    });
+describe('when required', () => {
+  it('should add `*` to title`', () => {
+    const { getByText } = renderMultilineTextField({ isRequired: true });
+    expect(getByText('*')).toBeInTheDocument();
```

Comment:
I guess this would pass if I have a title of * but not a isRequired passed. I would guess it is gine 😌

Reply:
Yeah it would, but the Visual Regression Test catches this anyways 🤷‍♂️